Correct conflict with Hl-line (this requires ov.el)

animike opened the pull request:
The Hl-line overlays were overriding the text background property of the face (in the highlighted line). This was because in Extrapolate the highlighting was being applied with a function that just changed the text properties and did not create overlays. Now by using the Ov package, overlays are applied.

animike committed:
Corrected conflict with Hl-line (requires ov.el)

The Hl-line overlays were overriding the text background property of the face (in the highlighted line). This was because in Extrapolate the highlighting was being applied with a function that only changed the text properties (according to the face) instead of creating an overlay to compete with Hl-line.

YoungFrog commented:
Thanks for checking.

I won't merge your pr because I don't like the name change (why do it
btw ?). OTOH I also don't care enough to argue or anything. So, since
you --understandably-- seem to care much more about this functionnality
than I do, I'll push a commit mentionning that you maintain your own
version on github. I hope this works for you. If there's anything I can
do to help, feel free to ask.

animike commented:
Thanks a lot! I thought of a different name because highlight-region was confusing to me because the functions had the same or too similar names to the mode and that resulted a bit difficult for me to follow the code. Also I thought that highlight-region was kind of a generic name and perhaps misleading because there are many functionalities or packages that highlight different things in different ways.

animike committed:
Added small delay to economize resources

No need to highlight for a too small fraction of time, but highlight fast enough to be not distracting.
